### PR TITLE
Make default partition behavior be a round-robin assignment when using Spring Cloud Azure EventHubs binder

### PR DESCRIPTION
**Pull request by Netyyyy**
As title.
Fixes https://github.com/Azure/azure-sdk-for-java/issues/32816

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinderTests.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.cloud.stream.provisioning.ProducerDestination;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import static org.mockito.Mockito.when;
+
+public class EventHubsMessageChannelBinderTests {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinderTests.java:24]: No `public`

**Comment by saragluna** [comment]
@moarychan please help review.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java [hunk 2]
```diff
@@ -117,8 +115,6 @@ protected MessageHandler createProducerMessageHandler(
         if (producerProperties.isPartitioned()) {
             handler.setPartitionIdExpression(
                 EXPRESSION_PARSER.parseExpression("headers['" + BinderHeaders.PARTITION_HEADER + "']"));
-        } else {
-            handler.setPartitionKeyExpression(new FunctionExpression<Message<?>>(m -> m.getPayload().hashCode()));
         }
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java:118]: Please check whether this change is necessary for the [Service Bus binder](https://github.com/Azure/azure-sdk-for-java/blob/cbbf4ba4b730c379c8b1091f2f1e3841ca323708/sdk/spring/spring-cloud-azure-stream-binder-servicebus/src/main/java/com/azure/spring/cloud/stream/binder/servicebus/ServiceBusMessageChannelBinder.java#L122).

**Comment by yiliuTo** [comment]
This would be a breaking change for users, so we can mark in in the changelog.

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinderTests.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Test
+    public void withPartitionConfig() {
+        producerProperties.setPartitionKeyExpression(new SpelExpressionParser().parseExpression("payload"));
+        MessageHandler handler = binder.createProducerMessageHandler(producerDestination,
+            producerProperties, errorChannel);
+
+        Assertions.assertNotNull(ReflectionTestUtils.getField(handler, PARTITION_ID_EXPRESSION));
+        Assertions.assertNull(ReflectionTestUtils.getField(handler, PARTITION_KEY_EXPRESSION));
+    }
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinderTests.java:65]: no public either

**Review by saragluna — APPROVED** [review]

**Pull request merged**